SmartDeviceLink/SDLCarWindowViewController.h

```diff
@@ -0,0 +1,15 @@
+//
+//  SDLCarWindowViewController.h
+//  SmartDeviceLink
+//
+//  Created by Joel Fischer on 12/12/17.
+//  Copyright © 2017 smartdevicelink. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface SDLCarWindowViewController : UIViewController
+
+@property (nonatomic, assign) UIInterfaceOrientation supportedOrientation;
+
```

Comment:
I tested the `SDLCarWindowViewController` with an app that supports all device orientations but with a view controller that only supported `landscapeRight`. This did not work because the view always loads initially in portrait mode must be rotated by the user in the `viewWillAppear` method. Once the view was rotated, the content disappeared off the screen and nothing could be streamed because the the screenshot returned a `nil` value. The only way I could get it to work was if the app only supports `landscapeRight` mode.

Reply:
So should we lock the orientation to portrait?

Reply:
I think setting the frame so part of the view controller is missing because it is off the screen is confusing from a UX standpoint. I think it would be better to force the orientation to landscape and then recalculate the `rootViewController`'s bounds so it matches the SDL accessory screen size but the entire view still fits on the screen. One negative is that this will complicate the `SDLTouchManager` because it means that it must scale touch points when the car window feature is being utilized.

Reply:
I don't think that will work. Accessory sizes could be in portrait (e.g. if someone made a screen like Tesla's) and could be of all sorts of resolutions. It would be more helpful to always have the lock screen displayed.

Reply:
We could just resize the `rootViewController` to best fit whichever the orientation the developer has chosen to lock the device. 

I think showing the lock screen all the time should not be the only option available to the developer. If a user wants to use the app on the phone while the car is parked we are inconveniencing them by making them pull and reconnect the USB cable multiple times.

Reply:
I want to note that this is _only_ an issue if the dev uses the root of the UIWindow to stream. If they use an off-screen VC it will work fine. That's the best practice. If the dev wants to stream the same VC being displayed on the phone, this _will never_ look right or be usable to the user. Developers shouldn't have to (and cannot) try and lock their app in an orientation based on the size of the connected screen.

Reply:
Good to know. The proposal is not very clear about how the car window API is supposed to be used so I assumed that it was meant to support both off-screen and visible screens.

Reply:
It does support visible screens, but there are some inherent limitations is how I would describe it.